cr.swift

```diff
@@ -0,0 +1,41 @@
+//
+//  Netguru iOS code review task
+//
+
+class paymentViewController: UIViewController {
+    var delegate: PaymentViewControllerDelegate
+    let customView = PaymentView()
+    let payment: Payment?
+
+    func callbacks() {
+        view.backgroundColor = UIColor(displayP3Red: 1, green: 1, blue: 1, alpha: 1)
+        customView.didTapButton = {
+            if let payment = self.payment {
+                self.delegate.didFinishFlow(amount: payment.amount
+                                            currency: payment.currency)
+            }
+        }
+    }
+
+    func viewDidLoad() {
+        navigationController?.setNavigationBarHidden(false, animated: false)
+        customView.centerXAnchor.constraint(equalTo: view.centerXAnchor)
+        customView.centerYAnchor.constraint(equalTo: view.centerYAnchor)
+        view.addSubview(customView)
+        fetchPayment()
+        setupCallbacks()
+    }
+
+    internal let ViewModel: PaymentViewModel
+
+    func fetchPayment() {
+        customView.statusText = "Fetching data"
+        ApiClient.sharedInstance().fetchPayment { payment in
```

Comment:
From the definition of the `sharedInstance` it seems to be a singleton, but singletons should be without `()`. It's hard to say in this case what exactly is wrong. Or `sharedInstance()` is not a singleton function, but object one or otherwise `()` must be removed. 

If it's object function, then naming is poor. In the second situation just remove `()`

In any case it's a very bad place to call this function here, as we already have `PaymentViewModel` for this type of logics and fetching data. `ViewController` should be responsible only for data presentation, but not for business logic or data fetching. 

Also I don't see any **error** type in response and how to present error if it will occur while API request. 

And of course last, but not least, we must catch `[weak self]` to prevent memory leaks